From 9f3eeb126143a510511789cf8669d8c24b99e086 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Thu, 26 May 2022 10:54:51 +1000
Subject: [PATCH] llvmpipe: flush resources for kms swrast path.

The kms_swrast path calls this callback via the dri2 paths,
not flushing caused artifacts when running inside a VM or on hw
in weston/gnome-shell.

Fixes: 6bbbe15a783a ("Reinstate: llvmpipe: allow vertex processing and fragment processing in parallel")
---
 src/gallium/drivers/llvmpipe/lp_surface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gallium/drivers/llvmpipe/lp_surface.c b/src/gallium/drivers/llvmpipe/lp_surface.c
index cc97565893d..78a1b06b81d 100644
--- a/src/gallium/drivers/llvmpipe/lp_surface.c
+++ b/src/gallium/drivers/llvmpipe/lp_surface.c
@@ -174,6 +174,7 @@ static void lp_blit(struct pipe_context *pipe,
 static void
 lp_flush_resource(struct pipe_context *ctx, struct pipe_resource *resource)
 {
+   llvmpipe_flush_resource(ctx, resource, 0, true, true, false, "resource");
 }
 
 
-- 
2.35.3

